specs/~dynamic-names.yml

```
@@ -139,6 +139,20 @@ tests:
     partials: { node: '{{content}}<{{#nodes}}{{>*template}}{{/nodes}}>' }
     expected: 'X<Y<>>'
 
+  - name: Dynamic Names - Dobule Dereferencing
+    desc: Dynamic Names can't be dereferenced more than once.
+    data: { dynamic: 'test', 'test': 'content' }
+    template: '"{{>**dynamic}}"'
+    partials: { content: 'Hello, world!' }
+    expected: '""'
+
+  - name: Dynamic Names - Composed Dereferencing
+    desc: Dynamic Names cannot have composed dereferencing.
+    data: { foo: 'fizz', bar: 'buzz', fizz: { buzz: { 'content' } } }
```

I think you also need a `fizz.bar -> 'buzz'`, as fallbacks to lower stack frames only apply to the first part of a dotted name.

```suggestion
    data: { foo: 'fizz', bar: 'buzz', fizz: { bar: 'buzz', buzz: { 'content' } } }
```

---

Good point. Should I make it become: `    data: { foo: 'fizz', bar: 'buzz', fizz: { bar: 'buzz', buzz: { 'content' } } }`?

---

I think that's what I suggested? Otherwise, I haven't been able to spot the difference yet.

---

sorry I meant `data: { foo: 'fizz', fizz: { bar: 'buzz', buzz: { 'content' } } }`